darwin/torch/dataset.py

```diff
@@ -141,96 +240,190 @@ def __getitem__(self, index: int):
+            sequence = convert_polygon_to_sequence(annotation["polygon"]["path"])
+            if len(sequence) / 2 < 3:
+                # Discard polygons with less than three points
+                continue
+            # Compute the bbox of the polygon
+            x_coords = sequence[0::2]
+            y_coords = sequence[1::2]
+            x = np.max((0, np.min(x_coords) - 1))
+            y = np.max((0, np.min(y_coords) - 1))
+            w = (np.max(x_coords) - x) + 1
+            h = (np.max(y_coords) - y) + 1
+            # Compute the area of the polygon
+            poly_area = polygon_area(x_coords, y_coords)
             bbox_area = w * h
             if poly_area > bbox_area:
                 raise ValueError(
                     f"polygon's area should be <= bbox's area. Failed {poly_area} <= {bbox_area}"
                 )
-            new_obj["area"] = poly_area
+            # Create and append the new entry for this annotation
+            target.append(
+                {
+                    "iscrowd": 0,
+                    "category_id": self.classes.index(annotation["name"]),
+                    "segmentation": [sequence],  # List type is used for backward compatibility
+                    "bbox": [x, y, w, h],
+                    "area": poly_area,
+                }
+            )
+
+        return {"image_id": index, "annotations": target}
+
+    def measure_weights(self, **kwargs):
+        """Computes the class balancing weights (not the frequencies!!) given the train loader
+        Get the weights proportional to the inverse of their class frequencies.
+        The vector sums up to 1
+
+        Returns
+        -------
+        class_weights : ndarray[double]
+            Weight for each class in the train set (one for each class) as a 1D array normalized
+        """
+        # Collect all the labels by iterating over the whole dataset
+        labels = []
+        for i, _ in enumerate(self.images_path):
+            target = self._map_annotation(i)
+            labels.extend([a["category_id"] for a in target["annotations"]])
+        return self._compute_weights(labels)
 
-            target.append(new_obj)
 
-        return {'image_id':index,
-                'annotations':target}
+####################################################################################################
 
 
 class SemanticSegmentationDataset(Dataset):
-    def __init__(self, dataset, split: Path, transforms: Optional[List] = None):
+    def __init__(self, root, split: Path, transform: Optional[List] = None):
         """See superclass for documentation"""
-        super().__init__(dataset=dataset, split=split, transforms=transforms)
-        self.classes = [e.strip() for e in open(str(self.dataset.local_path / "lists/classes_masks.txt"))]
+        super().__init__(root=root, split=split, transform=transform)
+        self.classes = [e.strip() for e in open(str(self.root / "lists/classes_polygon.txt"))]
```

Comment:
it's not good to call open like this, the file handler stays open (that's why we usually use `with open() as f:` since it automatically closes it.

But since `root` is already a path you can do this:

`[ e.strip() for e in (self.root / "lists/classes_polygon.txt").read_text().split("\n")]`

Reply:
done